srcalc/src/main/java/gov/va/med/srcalc/vista/RpcVistaPatientDao.java

```diff
@@ -223,25 +225,69 @@ private void parseRecentVitalResults(final Patient patient, final List<String> v
-        final List<String> bmiLineTokens = Splitter.on(compliedPattern).splitToList(vitalResults.get(7));
-        // The BMI value is the second to last token on its line
-        patient.setBmi(new RetrievedValue(
-            Double.parseDouble(bmiLineTokens.get(bmiLineTokens.size()-2)),
-            patient.getWeight().getMeasureDate(),
-            ""));
+        for(final String line: vitalResults)
+        {
+            // For each line, determine if we need to pull the vital information for that line
+            // Each entry comes with an optional date and time.
+            final Pattern pattern = Pattern.compile(
+                    "(.+:)\\s+" // Match to the first colon in order to get the name of the vital
+                    + "(\\(.+?\\))?\\s+" // Optional capture group for the date
+                    + "(.+)\\s+" // Match the value of the vital and any information between
+                                 // the optional date and the name
+                    + "([A-Za-z0-9_,]+)"); // Match the name of the person who recorded the vital
+            final Matcher matcher = pattern.matcher(line);
+            matcher.matches();
+            // Matcher groups begin at 1, as 0 denotes the entire pattern that was matched.
+            final String vitalName = matcher.group(1);
+            // This will be null when the date is not present.
+            final String dateString = matcher.group(2);
+            final String[] valuesArray = matcher.group(3).split(VALUES_SPLIT_REGEX);
+            switch(vitalName)
+            {
+                case "Ht.:":
+                    patient.setHeight(new RetrievedValue(
+                            findHeightInInches(valuesArray),
+                            dateFormat.parse(dateString),
+                            HEIGHT_UNITS));
+                    break;
+                case "Wt.:":
+                    patient.setWeight(new RetrievedValue(
+                            // Remove any asterisks in the weight value
+                            Double.parseDouble(valuesArray[0].replace("*", "")),
+                            dateFormat.parse(dateString),
+                            WEIGHT_UNITS));
+                    break;
+                case "Body Mass Index:":
+                    /* The actual name of this vital is "Body Mass Index:" but we are splitting
+                     * on whitespace so only test for the first word.
+                     * The BMI value is the second to last token on its line.
+                     */
+                    patient.setBmi(new RetrievedValue(
+                        // Remove any asterisks in the BMI value
+                        Double.parseDouble(valuesArray[0].replace("*", "")),
+                        patient.getWeight().getMeasureDate(),
+                        ""));
+                    break;
+            }
+        }
+    }
+    
+    private double findHeightInInches(final String[] valuesArray)
```

Comment:
`valuesArray` is not a very helpful name for this parameter. Perhaps `valuesWords`?